source/docs/hardware-reference/talonfx/improving-performance-with-current-limits.rst

```diff
@@ -0,0 +1,215 @@
+
+Supply, Stator and Torque Limits
+--------------------------------
+
+It's important to understand how current limits work. When a motor is under a load, it takes an increasing amount of current to continue rotating the shaft of the motor (and by extension the mechanism). When integrating multiple motors into a system (such as an FRC robot), this increase current draw can drain batteries, reset fuses, blow breakers, or even damage the battery.
+
+There exists 3 forms of current limiting: supply, stator and torque. For all limits, the following :ref:`configs <docs/tuner/configs:tuner configs>` apply. In the below example, replace ``Supply`` with ``Stator``.
+
+.. note:: Limits for torque based control modes should be applied with ``Peak Forward Torque Current`` and ``Peak Reverse Torque Current`` instead.
+
+- ``SupplyCurrentLimit``
+- ``SupplyCurrentLimitEnable``
+
+All limits must be enabled using the appropriate enable config.
+
+Supply Limits
+^^^^^^^^^^^^^
+
+.. note:: Supply limits are not functional in torque based control modes. Use the peak torque current configs instead.
+
+When ``SupplyCurrentThreshold`` has elapsed for ``SupplyTimeThreshold`` amount of time, it will lower motor output until it's within range of the current limit (see the above section on examples of these configs).
+
+Supply current limits are useful to prevent fuses from reseting and breakers from tripping. They are also effective at preventing brownouts and reducing overall current load on the battery.
+
+Stator & Torque Limits
+^^^^^^^^^^^^^^^^^^^^^^
+
+.. important::
+
+   Stator current limits are only applicable in non-torque control modes. For example, a stator limit would have no affect if the current control mode is ``TorqueCurrentFOC``. Users utilizing torque based control modes should use ``Peak Forward Torque Current`` and ``Peak Reverse Torque Current`` configs instead, but the following documentation is still applicable.
+
+Stator current is the output current (and proportional to torque) of the motor. The stator current differs from supply current in that it's at a different voltage from the input.
+
+.. note:: The below example ignores factors such as energy loss from heat or inefficiency.
+
+For example, if a motor is being applied with 50% dutycycle with :math:`80A`` of load (stator current), the resulting supply current will be :math:`40A`. If the motor has 100% dutycycle applied wth :math:`80A`` of stator current, the resulting supply current wll be :math:`80A`.
+
+.. figure:: images/stator-limit.png
+   :alt: 80a stator limit being applied
+
+   *80a torque limit taking affect*
+
+Stator limits apply the same limiting strategy used by supply limits, but are more effective in dealing with situations such as wheel slip or acceleration induced brownouts. The rationale behind it reducing acceleration induced brownouts is that initial acceleration is typically peak supply and stator current.
```

Comment:
I wouldn't mention "the same limiting strategy", it's only the same in that it limits motor output.